### PR TITLE
Backport Java-only Pubsub

### PR DESCRIPTION
**Pull request by mshields822**
https://github.com/apache/incubator-beam/pull/120
https://github.com/apache/incubator-beam/pull/213
https://github.com/apache/incubator-beam/pull/171
https://github.com/apache/incubator-beam/pull/346

### CODE REVIEW

**Comment by mshields822** [comment]
@dhalperi @davorbonaci 
NOT READY FOR REVIEW
But this is mostly it, stress testing tonight.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java [hunk 1]
```diff
@@ -1051,11 +1052,9 @@ public void translate(
     // IO Translation.
 
     registerTransformTranslator(
-        PubsubIO.Read.Bound.class, new PubsubIOTranslator.ReadTranslator());
+        PubsubUnboundedSource.class, new PubsubIOTranslator.ReadTranslator());
     registerTransformTranslator(
-        DataflowPipelineRunner.StreamingPubsubIOWrite.class,
-        new PubsubIOTranslator.WriteTranslator());
-
+        PubsubUnboundedSink.class, new PubsubIOTranslator.WriteTranslator());
```
Comment (mshields822) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: @peihe 
Pei, I could do with your help here. Despite my efforts these rewrites never seem  to apply when running on Google Cloud Dataflow.

Comment (peihe) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: PubsubIOTranslator.ReadTranslator is a TransformTranslator<PubsubIO.Read.Bound<T>>.
The type doesn't match for the changes above.

Are you trying to override the new pubsubIO back to nativeIO for Dataflow?

If so, I think you want to remove the two registerTransformTranslator above, and add the overide in DataflowPipelineRunner.
Here is how I did it for BigQuery,Write:
https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/205/files

**Comment by mshields822** [comment]
Ran beautifully overnight on dataflow + java pubsub (both sink and source).

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java (hunk 1 above)
Comment (mshields822) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: Thanks, I copied your style and it looks good.

**Comment by mshields822** [comment]
Confirmed works, porting back to BEAM.

**Comment by dhalperi** [comment]
Merged. Woohoo!

**Comment by mshields822** [comment]
Thanks!

**Pull request closed without merging**